filelog.go

```diff
@@ -87,6 +91,50 @@ func (w *FileLogWriter) Close() {
 	<- w.completed
 }
 
+// Track write failures and prints to stderr when possible. If err is nil, we'll try to clear the failures
+func (w *FileLogWriter) handleWriteFailure(err error) {
+	// Try to note any previous failures
+	if w.writeFailures != 0 {
+		_, fprintfErr := fmt.Fprintf(os.Stderr, "FileLogWriter(%q): Dropped %d previous log messages\n", w.filename, w.writeFailures)
+		if fprintfErr != nil {
+			// If we can't print now, exit early and try later
+			if err != nil { w.writeFailures += 1 }
+			return
+		} else {
+			w.writeFailures = 0
+		}
+	}
+	// If we have a current failure, attempt to print it
+	if err != nil {
+		_, fprintfErr := fmt.Fprintf(os.Stderr, "FileLogWriter(%q): Write failed: %v\n", w.filename, err)
+		if fprintfErr != nil {
+			w.writeFailures += 1
+		}
```

Comment:
Should we also reset `w.writeFailures` here if the right succeeds? Aren't we going to get an extra "Dropped previous messages" notice an extra time if not?